guaranteed-message-delivery/README.md

```diff
@@ -0,0 +1,442 @@
+
+        ![Define the resource](./resources/define_a_resource.gif)
+
+        The generated service will be as follows.
+
+        ```ballerina
+        service /healthcare on new http:Listener(8290) {
+            resource function post categories/[string category]/reserve(ReservationRequest request) 
+                    returns http:Created|http:InternalServerError {
+                
+            }
+        }
+        ```
+
+5. Define a [configurable variable](https://ballerina.io/learn/by-example/#configurability) for the name of the queue.
+
+    ![Define a configurable variable](./resources/define_a_configurable_variable.gif)
+
+    The generated code will be as follows.
+
+    ```ballerina
+    configurable string queueName = "ReservationQueue";
+    ```
+
+6. Create a [rabbitmq:Client](https://central.ballerina.io/ballerinax/rabbitmq/latest#Client) object to forward the message to the message broker.
+    
+    - Use RabbitMQ's default host and default port to initialize the client object.
+
+    ![Define a RabbitMQ client](./resources/define_a_rabbitmq_client.gif)
+
+    The generated code will be as follows.
+
+    ```ballerina
+    final rabbitmq:Client rabbitmqClient = check new (rabbitmq:DEFAULT_HOST, rabbitmq:DEFAULT_PORT)
+    ```
+
+7. Implement the logic.
+
+    ```ballerina
+    service /healthcare on new http:Listener(8290) {
+        function init() returns error? {
+            return rabbitmqClient->queueDeclare(queueName);
+        }
+
+        resource function post categories/[string category]/reserve(ReservationRequest request)
+                        returns http:Created|http:InternalServerError {
+            Patient patient = request.patient;
+            string doctor = request.doctor;
+            string hospital = request.hospital;
+
+            rabbitmq:Error? response = rabbitmqClient->publishMessage({
+                content: {
+                    patient,
+                    doctor,
+                    hospital_id: request.hospital_id,
+                    hospital: hospital,
+                    appointment_date: request.appointment_date,
+                    category
+                },
+                routingKey: queueName
+            });
+
+            if response is rabbitmq:Error {
+                log:printError("Failed to publish to the message broker", patient = patient.name,
+                                                                        doctor = doctor,
+                                                                        hospital = hospital);
+                return <http:InternalServerError>{body: response.message()};
+            }
+
+            return http:CREATED;
+        }
+    }
+    ```
+
+    - Define an `init` function to initialize the service. In there, declare the queue of the message-broker by calling `queueDeclare` function.
+
+    - In the resource function, first, define three variables and assign them values from the `request`.
```

Comment:
resource function -> resource method

I would update the latter part of the sentence to include more details about what the variables are for.

Also, would either use "from `request`" or "from the `request` record".